x-pack/plugins/security_solution/public/timelines/components/timeline/query_tab_content/index.tsx

```diff
@@ -346,51 +338,41 @@ export const QueryTabContentComponent: React.FC<Props> = ({
         refetch={refetch}
         skip={!canQueryTimeline}
       />
-      <FullWidthFlexGroup gutterSize="none">
+      <FullWidthFlexGroup gutterSize="s">
         <ScrollableFlexItem grow={2}>
           <StyledEuiFlyoutHeader
             data-test-subj={`${activeTab}-tab-flyout-header`}
             hasBorder={false}
           >
-            <EuiFlexGroup
-              alignItems="center"
-              gutterSize="s"
-              data-test-subj="timeline-date-picker-container"
-            >
+            <EuiFlexGroup gutterSize="s" direction="column">
               {timelineFullScreen && setTimelineFullScreen != null && (
-                <ExitFullScreen
-                  fullScreen={timelineFullScreen}
-                  setFullScreen={setTimelineFullScreen}
-                />
+                <EuiFlexItem>
+                  <EuiFlexGroup alignItems="center" gutterSize="s">
+                    <ExitFullScreen
+                      fullScreen={timelineFullScreen}
+                      setFullScreen={setTimelineFullScreen}
+                    />
+                  </EuiFlexGroup>
+                </EuiFlexItem>
               )}
-              <EuiFlexItem grow={10}>
-                <SuperDatePicker
-                  width="auto"
-                  id={InputsModelId.timeline}
-                  timelineId={timelineId}
-                  disabled={isDatePickerDisabled}
-                />
+              <EuiFlexItem data-test-subj="timeline-date-picker-container">
+                <QueryTabHeaderContainer data-test-subj="timelineHeader">
+                  <QueryTabHeader
+                    filterManager={filterManager}
+                    show={show && activeTab === TimelineTabs.query}
+                    showCallOutUnauthorizedMsg={showCallOutUnauthorizedMsg}
+                    status={status}
+                    timelineId={timelineId}
+                  />
+                </QueryTabHeaderContainer>
               </EuiFlexItem>
-              <EuiFlexItem grow={false}>
-                <TimelineDatePickerLock />
-              </EuiFlexItem>
-              <SourcererFlex grow={1}>
-                {activeTab === TimelineTabs.query && (
-                  <Sourcerer scope={SourcererScopeName.timeline} />
-                )}
-              </SourcererFlex>
+              {/* TODO: This is a temporary solution to hide the KPIs until lens components play nicely with timelines */}
+              {/* https://github.com/elastic/kibana/issues/17156 */}
+              {/* <EuiFlexItem grow={false}> */}
+              {/*   <TimelineKpi timelineId={timelineId} /> */}
+              {/* </EuiFlexItem> */}
```

Comment:
🧹

Reply:
@semd We're probably going to leave this for the time being as Paul wants this hidden for now until we integrate lens. If you think we should still remove it out of the code we can reference this commit and line in an issue documented here?

Reply:
Oh, that's okay then. no problem.

Reply:
I would consider deleting the commented code. I mentioned it somewhere else but there are a lot of new files introduced by the PR that I don't think should be added either. It introduces confusion (especially for the new files) and they might be completely changed in the future...

Reply:
@logeekal Could this snippet be extracted to a comment in the corresponding Github issue as a reminder, and clean this? I tend to prefer this way to leaving comments in the code, most of them stay there forever.

@PhilippeOberti I understand, would you be okay if we let this PR go through, and then you create a separate PR refactoring the implementation to make it less confusing?

Reply:
I will not block merging this PR because of it, but really think it should be removed. It's just really bad practice to merge code that we know isn't used. These few commented lines aren't the biggest problem here, for me all the new kpis files are more annoying. They will appear in the git history for potentially no reason at all if we end up not using them and rewriting them entirely!

Reply:
The rational for keeping these files were that these were based on the new designs and decision to exclude them temporarily was taken at last minute.

If I do not leave the comment there and somebody resumes KPIs on similar designs, they will be starting from scratch which I don't want. 

But I am okay removing them as well, if we want to talk more about it or that is the consensus.